generate/newlicense.go

```diff
@@ -37,6 +37,9 @@ type NewLicenseInfo struct {
 	TLSKeyPair authority.TLSKeyPair
 	// Payload is the license payload
 	Payload []byte
+	// CreateAnonymizationKey specifies whether a key should be created
+	// for this license to anonymize usage data
+	CreateAnonymizationKey bool
```

Comment:
We definitely need to be able to parse the anonymization key with this library - that code looks great. Go makes this almost too easy.

I'm not sure about _creating_ the anonymization key here though.

Personally, I'm okay with it so long as we make sure the license with the anonymization key isn't what gets stored to the database and that we don't retain it.

We would be able to provide even stronger guarantees if this library could only read (but not generate) the anonymization key, and then the anonymization key could be generated on-demand (and appended to the license.pem that we already have) when the user clicks the button to download the license.

What do you think @klizhentas?

Reply:
Thanks!

So @Envek that means you have 2 options:

1. Generate the anon key at the time you generate the license (which is what you have here). You'll have to make sure that you remove the anon key before you store the license in the database, because the whole point of these scheme is that we don't retain this key and can't leverage it to de-anonymize customer data.
2. Generate the license without the anon key, and append the anon key after the license is generated and stored in the database.

On the surface, #2 seems less prone to errors, but totally up to you.

Reply:
Agree that it is better to do it by option 2: append the anonymization key to license file after it has been saved to the database without it.

Maybe it worth to keep anonymization key generation/append logic in this library in separate functions, that will be called by the code that prepares/sends license to the user.